static/app/components/charts/intervalSelector.tsx

```diff
@@ -202,27 +214,37 @@ export default function IntervalSelector({
         )
       );
     }
-    return filteredResults;
+
+    return filteredResults.map(option => ({
+      ...option,
+      textValue: option.value,
+    }));
   };
 
+  if (!INTERVAL_DISPLAY_MODES.includes(displayMode)) {
+    return null;
+  }
+
   return (
-    <DropdownAutoComplete
-      onSelect={item => onIntervalChange(item.value)}
-      items={intervalOption.options.map(option => ({
-        value: option,
-        searchKey: option,
-        label: option,
-      }))}
+    <CompactSelect
       searchPlaceholder={t('Provide a time interval')}
-      autoCompleteFilter={(items, filterValue) =>
-        intervalAutoComplete(items, filterValue)
-      }
-    >
-      {({isOpen}) => (
-        <DropdownButton borderless prefix={t('Interval')} isOpen={isOpen}>
-          {interval}
-        </DropdownButton>
-      )}
-    </DropdownAutoComplete>
+      value={interval}
+      options={items}
+      onChange={option => onIntervalChange(option.value)}
+      searchable
+      disableSearchFilter
+      onSearch={filterValue => {
+        setItems(intervalAutoComplete(filterValue));
+      }}
```

Comment:
built-in filtering of items can’t be used because the filter options are expected to change while typing:

![Screenshot 2025-04-08 at 13 41 55](https://github.com/user-attachments/assets/a47ad7be-4b2f-4d56-8a5a-52be75af2bfe)
![Screenshot 2025-04-08 at 13 42 03](https://github.com/user-attachments/assets/03b8cc66-d386-4e06-ac90-4c3d7957bdda)

for this, we need to control searching from the outside with state